spanner/errors.go

```diff
@@ -104,10 +110,13 @@ func (e *Error) decorate(info string) {
 }
 
 // spannerErrorf generates a *spanner.Error with the given description and a
-// status error with the given error code as its wrapped error.
+// APIError error having given error code as its status.
 func spannerErrorf(code codes.Code, format string, args ...interface{}) error {
 	msg := fmt.Sprintf(format, args...)
 	wrapped := status.Error(code, msg)
+	if apiError, ok := apierror.FromError(wrapped); ok {
```

Comment:
This check should really never fail, and if it would fail, we would start returning errors that are no longer in accordance with the documentation. We should therefore have a test that verifies that this does not fail, so we do not accidentally release a version that is not able to create an `APIError` from a `status.Error`.

Put another way: I think we should remove the `if` statement and just assume that it was ok, and have a test case that checks that the error that is returned from `spannerErrorf` always wraps an `APIError`.